swaylock/main.c

```diff
@@ -156,7 +157,64 @@ static const struct wl_registry_listener registry_listener = {
 
 static struct swaylock_state state;
 
+bool create_file_lock() {
```

Comment:
Why? If the user starts a lock screen, and leaves the sceen, then swayidle will try to start a lockscreen and get an error. I prefer handling known race scenarious and exit cleanly. I will update this to include the WAYLAND_DISPLAY in the path, so you can run lockscreens in paralell sway instances.

Reply:
Including `WAYLAND_DISPLAY` is a little bit better.

Why not having the compositor do this test? The compositor knows if the swaylock is running, because it started it.

Reply:
The compositor will use the already_inhibited error on input-inhibitor, which aborts any swaylock clients after the first.